tests/foreman/cli/test_classparameters.py

```diff
@@ -31,68 +31,54 @@
-        """Import some parametrized puppet classes. This is required to make
-        sure that we have smart class variable available.
-        Read all available smart class parameters for imported puppet class to
-        be able to work with unique entity for each specific test.
-        """
-        super().setUpClass()
-        cls.puppet_modules = [{'author': 'robottelo', 'name': 'cli_test_classparameters'}]
-        cls.org = make_org()
-        cls.loc = make_location()
-        cv = publish_puppet_module(cls.puppet_modules, CUSTOM_PUPPET_REPO, cls.org['id'])
-        cls.env = Environment.list({'search': 'content_view="{}"'.format(cv['name'])})[0]
-        Environment.update(
-            {
-                'name': cls.env['name'],
-                'organization-ids': cls.org['id'],
-                'location-ids': cls.loc['id'],
-            }
-        )
-        cls.puppet_class = Puppet.info(
-            {'name': cls.puppet_modules[0]['name'], 'environment': cls.env['name']}
-        )
-        cls.sc_params_list = SmartClassParameter.list(
-            {
-                'environment': cls.env['name'],
-                'search': 'puppetclass="{}"'.format(cls.puppet_class['name']),
-            }
-        )
-        cls.sc_params_ids_list = [sc_param['id'] for sc_param in cls.sc_params_list]
-        cls.host = entities.Host(
-            organization=cls.org['id'], location=cls.loc['id'], environment=cls.env['name']
-        ).create()
-        cls.host.add_puppetclass(data={'puppetclass_id': cls.puppet_class['id']})
-        cls.hostgroup = make_hostgroup(
-            {'environment-id': cls.env['id'], 'puppet-class-ids': cls.puppet_class['id']}
-        )
 
-    @classmethod
-    def tearDownClass(cls):
-        """Removes puppet class."""
-        super().tearDownClass()
-        delete_puppet_class(cls.puppet_class['name'])
+@pytest.fixture(scope='module')
+def module_loc():
+    return make_location()
 
-    def setUp(self):
-        """Checks that there is at least one not overridden
-        smart class parameter before executing test.
-        """
-        super().setUp()
-        if len(self.sc_params_list) == 0:
-            raise Exception("Not enough smart class parameters. Please update puppet module.")
+
+@pytest.fixture(scope='module')
+def module_puppet(module_org, module_loc):
+    puppet_modules = [{'author': 'robottelo', 'name': 'cli_test_classparameters'}]
+    cv = publish_puppet_module(puppet_modules, CUSTOM_PUPPET_REPO, module_org['id'])
+    env = Environment.list({'search': f'content_view="{cv["name"]}"'})[0]
+    Environment.update(
+        {
+            'name': env['name'],
+            'organization-ids': module_org['id'],
+            'location-ids': module_loc['id'],
+        }
+    )
+    puppet_class = Puppet.info({'name': puppet_modules[0]['name'], 'environment': env['name']})
+    yield {'modules': puppet_modules, 'env': env, 'class': puppet_class}
+    delete_puppet_class(puppet_class['name'])
```

Comment:
you can remove this

Reply:
I would rather keep it to be able to rerun some of the tests against the same instance. Otherwise we would get "Could not create the override value:\n  Validation failed: Match has already been taken".